docs/architecture/posixfs.md

```diff
@@ -0,0 +1,173 @@
+
+All indexing and caching of metadata is implemented in higher system levels than the storage driver, and thus are not different to the components used with other storage drivers like the decomposed FS.
+
+### Monitoring
+
+To get information about changes such as new files added, files edited or removed, Infinte Sale uses a monitoring system to directly watch the file system. This starts with the Linux inotify system and ranges to much more sophisticated services as for example in Spectrum Scale (see [GPFS Specifics](#gpfs-specifics) for more details on GPFS file systems).
+
+Based on the information transmitted by the watching service, Infinite Scale is able to "register" new or changed files into its own caches and internal management structures. This enables Infinite Scale to deliver resource changes through the "traditional" channels such as APIs and clients.
+
+Since the most important metadata is the file tree structure itself, the "split brain" situation between data and metadata is impossible to cause trouble.
+
+### Automatic ETag Propagation
+
+The ETag of a resource can be understood as a content fingerprint of any file- or folder resource in Infinite Scale. It is mainly used by clients to detect changes of resources. The rule is that if the content of a file changed the ETag has to change as well, as well as the ETag of all parent folders up to the root of the space.
+
+Infinite Scale uses a built in mechanism to maintain the ETag for each resource in the file meta data, and also propagates it automatically.
+
+In the future a sophisticated underlying file system could provide an attribute that fulfills this requirement and changes whenever content or metadata of a resource changes, and - which is most important - also changes the attribute of the parent resource and the parent of the parent etc.
+
+### Automatic Tree Size Propagation
+
+Similar to the ETag propagation described before, Infinite Scale also tracks the accumulated tree size in all nodes of the file tree. A change to any file requires a re-calculation of the size attribute in all parent folders.
+
+In the future Infinite Scale could benefit from file systems with native tree size propagation.
+
+### Quota
+
+Each space has it's own quota, thus every storage driver implementation needs to consider that.
+
+For example, IBM Spectrum Scale supports quota handling directly in the file system.
+
+Other systems store quota data in the metadata storage and implement propagation of used quota similar to the tree size propagation.
+
+### File Id Resolution
+
+Infinite Scale uses an Id based approach to work with resources, rather than a file path based mechanism. The reason for that is that Id based lookups can be done way more efficient compared to tree traversals, just to name one reason.
```

Comment:
```suggestion
Infinite Scale uses an Id based approach to work with resources, rather than a file path based mechanism. The reason for that is that Id based lookups can be done way more efficiently compared to tree traversals, just to name one reason.
```